### PR TITLE
Fix serviceinfo nil case

### LINKED ISSUE #5786: [production][package#show] undefined method `[]' for nil:NilClass
package#show threw this 1 times
<pre>
undefined method `[]&#39; for nil:NilClass
</pre>
[https://build.opensuse.org/package/show/home:duwe:branches:Kernel:HEAD/kernel-default](https://build.opensuse.org/package/show/home:duwe:branches:Kernel:HEAD/kernel-default)"

[See this exception on Errbit](https://errbit-opensuse.herokuapp.com/apps/5620ca2bdc71fa00b1000000/problems/5b8f8beefe4eac0014218d70)

### PR DESCRIPTION
**Pull request by saraycp**
Return an empty hash if serviceinfo is missing.

Closes #5787
Fixes #5786

Co-authored-by: David Kang <dkang@suse.com>
Co-authored-by: Moisés Déniz Alemán <mdeniz@suse.com>

### CODE REVIEW

File: src/api/spec/models/package_spec.rb [hunk 1]
```diff
@@ -230,6 +230,19 @@
     end
   end
 
+  describe '#serviceinfo' do
+    let(:url) { "#{CONFIG['source_url']}/source/#{package_with_service.project}/#{package_with_service.name}" }
+    let(:no_serviceinfo) do
+      '<directory name="package_with_service" rev="1" vrev="1" srcmd5="cf9c84e27a27dfc3e289f74fb096b42a">
```
Comment (coolo) [inline src/api/spec/models/package_spec.rb]: you can do multiline strings like this? And why is everyone else doing the <<~XML dance? I feel underinformed!! :)

File: src/api/app/models/package.rb [hunk 2]
```diff
@@ -1251,7 +1251,7 @@ def self.is_binary_file?(filename)
   def serviceinfo
     begin
       dir = Directory.hashed(project: project.name, package: name)
-      return dir['serviceinfo'] if dir
+      return dir['serviceinfo'] || {} if dir
```
Comment (coolo) [inline src/api/app/models/package.rb]: perhaps ``dir.fetch('serviceinfo', {})`` is a little easier to read in this context?  I find the || combined with the if a little hard - but nitpicking

File: src/api/spec/models/package_spec.rb (hunk 1 above)
Comment (mdeniz) [reply src/api/spec/models/package_spec.rb]: We just C&P from the spec for service_error...

Comment (hennevogel) [reply src/api/spec/models/package_spec.rb]: I like `file_fixture` best...

Comment (bgeuken) [reply src/api/spec/models/package_spec.rb]: I don't. They hide what we test and make the test harder to read. For oneliner strings (shorter strings in general) I would prefer the heredoc syntax

Comment (saraycp) [reply src/api/spec/models/package_spec.rb]: If we achieve an agreement on that, I'd do it in a new PR and refactor the whole file.
I have no strong opinion, but I think I'd prefer `file_fixture`... :thinking:

Comment (coolo) [reply src/api/spec/models/package_spec.rb]: I'm with @bgeuken - the extra file makes it harder

Comment (hennevogel) [reply src/api/spec/models/package_spec.rb]: > If we achieve an agreement on that

We don't need to. This was just a conversation :smile:

**Pull request merged**